### PR TITLE
Privatemsg improvements for D7

### LINKED ISSUE #497: D7: Port or replace ws_pm_email_notify
ws_pm_email_notify is a fork of pm_email_notify module, a submodule of privatemsg.

It's quite probable that it didn't need to be forked and that its functionality can be implemented as hooks on an unmodified pm_email_notify (that same module still exists on the D7 7.x-1.x branch of privatemsg). 

The two key features of this module from our perspective:

1. Sending out email notifications that have response links and correct reply-to and other details
2. Handling incoming emails (via mandrill_incoming module)

Testing of mandrill incoming stuff is a little complicated, as it requires a publicly accessible port.  I [wrote an article about tunnelling external webservice call](http://randyfay.com/content/tunneling-http-debug-or-develop-external-webservice-call) when originally doing this stuff to explain how to set up a development and testing environment.

Prerequisite: [D7 version of mandrill_incoming](https://github.com/warmshowers/Warmshowers.org/issues/496).

### PR DESCRIPTION
**Pull request by rfay**
- Create ws_pm_email_notify module, which alters outgoing mail to add our tracking and routing.
- Implement hook_mandrill_incoming() to manage incoming mandrill mail.
- Migrate pm_email_notify translations from D6 into variables feature
- Convert all messages to D7 tokens
- Patch D7 privatemsg to use hook_variable_info() and send properly localized messages.

Fixes #497 "Port or replace pm_email_notify"

### CODE REVIEW

File: docroot/sites/all/modules/patched/privatemsg/patches/privatemsg.pm_email_notify_variable_based_localized_email_2505495_01.patch [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+       drupal_mail('pm_email_notify', 'notice', $recipient->mail, user_preferred_language($recipient), $params, !empty($from) ? $from : NULL);
+     }
+   }
+@@ -65,6 +66,14 @@ function pm_email_notify_mail($key, &$message, $params) {
+         'privatemsg_message' => $params['message'],
+         'privatemsg_recipient' => $params['recipient'],
+       );
++
++      $subject = variable_get('pm_email_notify_subject', 'New private message at [site:name].');
++      $body = variable_get('pm_email_notify_body', 'New private message at [site:name].');
```
Comment (kalpaitch) [inline docroot/sites/all/modules/patched/privatemsg/patches/privatemsg.pm_email_notify_variable_based_localized_email_2505495_01.patch]: I'm just being super fussy but is there a need for two variable get calls (for each variable) if localization module exists, could we not put the default variable get below in an else... #savingtheworldonecallatatime

File: docroot/sites/all/modules/patched/privatemsg/patches/privatemsg.pm_email_notify_variable_based_localized_email_2505495_01.patch [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
++      if (function_exists('i18n_variable_get')) {
++        $subject = i18n_variable_get('pm_email_notify_subject', user_preferred_language($params['recipient'])->language, 'New private message at [site:name].');
++        $body = i18n_variable_get('pm_email_notify_body', user_preferred_language($params['recipient'])->language, _pm_email_notify_default_body());
++      }
```
Comment (kalpaitch) [inline docroot/sites/all/modules/patched/privatemsg/patches/privatemsg.pm_email_notify_variable_based_localized_email_2505495_01.patch]: I'm just being super fussy but is there a need for two variable get calls (for each variable) if localization module exists, could we not put the default variable get below in an else... #savingtheworldonecallatatime

Comment (rfay) [inline docroot/sites/all/modules/patched/privatemsg/patches/privatemsg.pm_email_notify_variable_based_localized_email_2505495_01.patch]: I've used this modality for decades now, and it is ever so slightly less efficient in some cases, but it accomplishes three things. 1) Makes sure variable is initialized, 2) Shows the "normal" or most common flow in the code, 3) Avoids unnecessary and hard-to-follow "else".  So I like it. It's a tiny bit less efficient, but in this context performance is completely irrelevant.

Comment (rfay) [inline docroot/sites/all/modules/patched/privatemsg/patches/privatemsg.pm_email_notify_variable_based_localized_email_2505495_01.patch]: Oh, but I should have started with "Thanks for the careful review!". It's ever so much appreciated that you read this carefully, and made a completely relevant and correct statement. Thanks!

**Pull request merged**